zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java

```diff
@@ -260,10 +616,66 @@ public String login(AuthenticationType authType) {
     public String getAccountUsername() {
         HAccount authenticatedAccount =
                 ServiceLocator.instance().getInstance(
-                        JpaIdentityStore.AUTHENTICATED_USER, HAccount.class);
+                        ZanataJpaIdentityStore.AUTHENTICATED_USER, HAccount.class);
         if (authenticatedAccount != null) {
             return authenticatedAccount.getUsername();
         }
         return null;
     }
+
+    public Principal getPrincipal() {
+        return principal;
+    }
+
+    public void checkRestriction(String expression) {
+        log.warn("permission check as EL is not supported yet");
+        throw new UnsupportedOperationException("identity.checkRestriction(exp) is not supported");
+    }
+
+    // copied from org.jboss.seam.security.Identity
+    @Override
+    public synchronized void runAs(RunAsOperation operation) {
+        Principal savedPrincipal = getPrincipal();
+        Subject savedSubject = getSubject();
+
+        try {
+            principal = operation.getPrincipal();
+            subject = operation.getSubject();
+
+            if (systemOp == null) {
+                systemOp = new ThreadLocal<>();
+            }
+
+            systemOp.set(operation.isSystemOperation());
+
+            operation.execute();
+        } finally {
+            systemOp.set(false);
```

Comment:
I wonder if this ThreadLocal could leak memory, in an environment where worker threads weren't reused.  Do you think it would be safer to call `systemOp.remove()`?